libraries/openssl/CMakeLists.txt

```diff
@@ -138,32 +138,34 @@ function(zeekAgentLibrariesOpenSSL)
     message(FATAL_ERROR "Unsupported system")
   endif()
 
-  add_library(thirdparty_openssl_libcrypto STATIC IMPORTED GLOBAL)
-  set_target_properties(thirdparty_openssl_libcrypto PROPERTIES
+  add_library(thirdparty_openssl_libssl STATIC IMPORTED GLOBAL)
+  set_target_properties(thirdparty_openssl_libssl PROPERTIES
     INTERFACE_INCLUDE_DIRECTORIES
       "${install_directory}/include"
 
     IMPORTED_LOCATION
-      "${install_directory}/${lib_folder_name}/libcrypto.${lib_extension}"
+      "${install_directory}/${lib_folder_name}/libssl.${lib_extension}"
   )
 
-  add_dependencies(thirdparty_openssl_libcrypto
+  add_dependencies(thirdparty_openssl_libssl
     thirdparty_openssl_builder
   )
 
-  add_library(thirdparty_openssl_libssl STATIC IMPORTED GLOBAL)
-  set_target_properties(thirdparty_openssl_libssl PROPERTIES
+  add_library(thirdparty_openssl_libcrypto STATIC IMPORTED GLOBAL)
+  set_target_properties(thirdparty_openssl_libcrypto PROPERTIES
     INTERFACE_INCLUDE_DIRECTORIES
       "${install_directory}/include"
 
     IMPORTED_LOCATION
-      "${install_directory}/${lib_folder_name}/libssl.${lib_extension}"
+      "${install_directory}/${lib_folder_name}/libcrypto.${lib_extension}"
   )
 
-  add_dependencies(thirdparty_openssl_libssl
+  add_dependencies(thirdparty_openssl_libcrypto
     thirdparty_openssl_builder
   )
 
+  target_link_libraries(thirdparty_openssl_libcrypto thirdparty_openssl_libssl)
```

Comment:
This may get lost when properties are propagated; the best place to add this is inside the thirdparty_openssl interface target as shown in issue #71 

The link chain should be something like:
`zeek agent -> thirdparty_openssl -> thirdparty_openssl_crypto | thirdparty_openssl_libssl`

`target_link_libraries(thirdparty_openssl INTERFACE libssl libcrypto)` will update INTERFACE_LINK_LIBRARIES in thirdparty_openssl and cause CMake to append libssl + libcrypto (in the stated order) inside the LINK_LIBRARIES property of the next target (zeek-agent).

Linking one imported target to the other (with either method, interface/private/public) will only update the next imported target without changing the INTERFACE_LINK_LIBRARIES property of thirdparty_openssl, causing the Zeek Agent to still see the original library order

Reply:
@bbannier any thoughts on this before I merge this?

Reply:
Does this work on your system? I am asking since the systems in CI never saw this issue.

Reply:
No this is not working on my system. I am getting errors while configuring it.
```
vagrant@host2:~/projects/zeek-agent/build$  cmake -DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo -DZEEK_AGENT_ENABLE_INSTALL:BOOL=ON -DZEEK_AGENT_ENABLE_TESTS:BOOL=ON -DZEEK_AGENT_ZEEK_COMPATIBILITY:STRING="3.1" /home/vagrant/projects/zeek-agent/
-- zeek-agent: Building with Zeek >= 3.1 compatibility
-- The C compiler identification is Clang 10.0.0
-- The CXX compiler identification is Clang 10.0.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- zeek-agent: ccache enabled (system default)
-- zeek-agent: Sanitizers are disabled
-- zeek-agent: Tests are enabled
-- Disabling the SQLite R*Tree module
-- Disabling the SQLite FTS extensions
CMake Error at libraries/openssl/CMakeLists.txt:167 (target_link_libraries):
  IMPORTED library can only be used with the INTERFACE keyword of
  target_link_libraries
Call Stack (most recent call first):
  libraries/openssl/CMakeLists.txt:176 (zeekAgentLibrariesOpenSSL)
```

Reply:
The custom commands that start the OpenSSL build are configured with a DEPENDS attribute that will automatically start the external build process if the required files (the static libraries) are missing. The build cache configured in GitHub Actions is restoring the libssl/libcrypt static libraries, so the build for thirdparty_openssl is not triggered again.

Non-interface properties such as LINK_LIBRARIES and INCLUDE_DIRECTORIES are blocked on imported targets, because they are not built by CMake (and can only propagate properties to dependencies). When setting target-based properties without any access specifier (which is deprecated) it defaults to PRIVATE, which in this case is not valid.

Force-pushing a single commit with the fix written by @Wajihulhassan should solve the issue:

```
--- a/libraries/openssl/CMakeLists.txt
+++ b/libraries/openssl/CMakeLists.txt
@@ -166,8 +166,8 @@ function(zeekAgentLibrariesOpenSSL)

   add_library(thirdparty_openssl INTERFACE)
   target_link_libraries(thirdparty_openssl INTERFACE
-    thirdparty_openssl_libcrypto
     thirdparty_openssl_libssl
+    thirdparty_openssl_libcrypto
   )
 endfunction()
```